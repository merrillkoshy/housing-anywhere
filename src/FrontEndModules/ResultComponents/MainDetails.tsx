import { JsxElement, parseJsonText } from "typescript";

interface Searchable {
	[key: string]: string;
}
const MainDetails = ({
	alias,
	currentoccupancy,
	deposit,
	estimatedbills,
	externalreference,
	extradata,
	kind,
	maxage,
	maxbookabledays,
	minage,
	minimumstaymonths,
	moveinwindow,
	preferredgender,
	type,
}: {
	type: string;
	kind: string;
	deposit?: number;
	estimatedbills?: number;
	minimumstaymonths: string;
	maxbookabledays?: number;
	moveinwindow: number;
	currentoccupancy: number;
	minage: number;
	maxage: number;
	preferredgender: string;
	alias: string;
	externalreference: string;
	extradata: string;
}) => {
	const dataExtracter = (stringData: string) => {
		const result = stringData.replaceAll("'", "").replaceAll("\\", "");
		return result;
	};
	const ExtraData = () => {
		const toObject = dataExtracter(extradata);
		const object = JSON.parse(toObject);
		for (const key in object) {
			if (Object.prototype.hasOwnProperty.call(object, key)) {
				const element = key;
				const value = object[key];
				return (
					<div className="col mx-2">
						<strong>{element}:</strong> {value}
					</div>
				);
			} else {
				<></>;
			}
		}
		return <></>;
	};

	return (
		<div className="d-flex flex-wrap w-100 justify-content-stretch">
			<div className="col mx-2">
				<strong>Type:</strong>{" "}
				{type === "1"
					? "House"
					: type === "2"
					? "Building"
					: type === "3"
					? "Apartment"
					: ""}
			</div>
			<div className="col mx-2">
				<strong>Kind:</strong>{" "}
				{kind === "1"
					? "Entire Place"
					: kind === "2"
					? "Private Room"
					: kind === "3"
					? "Shared"
					: ""}
			</div>

			<div className="col mx-2">
				<strong>Deposit:</strong> {deposit}
			</div>
			<div className="col mx-2">
				<strong>Estimated Bills:</strong> {estimatedbills}
			</div>
			<div className="col mx-2">
				<strong>Minimum Stay:</strong> {minimumstaymonths} Months
			</div>
			<div className="col mx-2">
				<strong>Maximum Bookbale Days:</strong> {maxbookabledays} days
			</div>
			<div className="col mx-2">
				<strong>Move-In Window:</strong> {moveinwindow} days
			</div>
			<div className="col mx-2">
				<strong>Current Occupants:</strong> {currentoccupancy}
			</div>
			<div className="col mx-2">
				<strong>Min Age:</strong> {minage}
			</div>
			<div className="col mx-2">
				<strong>Max Age:</strong> {maxage}
			</div>
			<div className="col mx-2">
				<strong>Preferred Gender:</strong> {preferredgender}
			</div>
			<div className="col mx-2">
				<strong>Alias:</strong> {alias}
			</div>
			<div className="col mx-2">
				<strong>External Reference:</strong> {externalreference}
			</div>
			<ExtraData />
		</div>
	);
};
export default MainDetails;
