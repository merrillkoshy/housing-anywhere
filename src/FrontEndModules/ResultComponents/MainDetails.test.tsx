import React from "react";
import { shallow } from "enzyme";
import MainDetails from "./MainDetails";

describe("MainDetails Costs module", () => {
	const MainDetailsComponent = shallow(
		<MainDetails
			type={""}
			kind={""}
			minimumstaymonths={""}
			moveinwindow={0}
			currentoccupancy={0}
			minage={0}
			maxage={0}
			preferredgender={""}
			alias={""}
			externalreference={""}
			extradata={""}
		/>
	);
	it("renders", () => {
		expect(MainDetailsComponent).toMatchSnapshot();
	});

	it("finds an li type", () => {
		expect(MainDetailsComponent.find("#type").exists()).toBe(true);
	});
	it("finds an li kind", () => {
		expect(MainDetailsComponent.find("#kind").exists()).toBe(true);
	});
	it("finds an li minimumstaymonths", () => {
		expect(MainDetailsComponent.find("#minimumstaymonths").exists()).toBe(true);
	});
	it("finds an li moveinwindow", () => {
		expect(MainDetailsComponent.find("#moveinwindow").exists()).toBe(true);
	});
	it("finds an li currentoccupancy", () => {
		expect(MainDetailsComponent.find("#currentoccupancy").exists()).toBe(true);
	});
	it("finds an li minage", () => {
		expect(MainDetailsComponent.find("#minage").exists()).toBe(true);
	});
	it("finds an li maxage", () => {
		expect(MainDetailsComponent.find("#maxage").exists()).toBe(true);
	});
	it("finds an li preferredgender", () => {
		expect(MainDetailsComponent.find("#preferredgender").exists()).toBe(true);
	});
	it("finds an li alias", () => {
		expect(MainDetailsComponent.find("#alias").exists()).toBe(true);
	});
	it("finds an li externalreference", () => {
		expect(MainDetailsComponent.find("#externalreference").exists()).toBe(true);
	});
});
