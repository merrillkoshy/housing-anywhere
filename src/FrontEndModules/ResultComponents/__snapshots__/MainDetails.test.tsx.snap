// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainDetails Costs module renders 1`] = `
<div
  className="mt-5 d-flex flex-wrap w-100 justify-content-stretch"
>
  <div
    className="d-flex flex-wrap col m-2  main-details"
  >
    <strong
      id="type"
    >
      Type:
    </strong>
     
  </div>
  <div
    className="d-flex flex-wrap col m-2  main-details"
  >
    <strong
      id="kind"
    >
      Kind:
    </strong>
     
  </div>
  <div
    className="d-flex flex-wrap col m-2  main-details"
  >
    <strong
      id="deposit"
    >
      Deposit:
    </strong>
     
  </div>
  <div
    className="d-flex flex-wrap col m-2  main-details"
  >
    <strong
      id="estimatedbills"
    >
      Estimated Bills:
    </strong>
     
  </div>
  <div
    className="d-flex flex-wrap col m-2  main-details"
  >
    <strong
      id="minimumstaymonths"
    >
      Minimum Stay:
    </strong>
     
     Months
  </div>
  <div
    className="d-flex flex-wrap col m-2  main-details"
  >
    <strong
      id="maxbookabledays"
    >
      Maximum Bookbale Days:
    </strong>
     
     days
  </div>
  <div
    className="d-flex flex-wrap col m-2  main-details"
  >
    <strong
      id="moveinwindow"
    >
      Move-In Window:
    </strong>
     
    0
     days
  </div>
  <div
    className="d-flex flex-wrap col m-2  main-details"
  >
    <strong
      id="currentoccupancy"
    >
      Current Occupants:
    </strong>
     
    0
  </div>
  <div
    className="d-flex flex-wrap col m-2  main-details"
  >
    <strong
      id="minage"
    >
      Min Age:
    </strong>
     
    0
  </div>
  <div
    className="d-flex flex-wrap col m-2  main-details"
  >
    <strong
      id="maxage"
    >
      Max Age:
    </strong>
     
    0
  </div>
  <div
    className="d-flex flex-wrap col m-2  main-details"
  >
    <strong
      id="preferredgender"
    >
      Preferred Gender:
    </strong>
     
  </div>
  <div
    className="d-flex flex-wrap col m-2  main-details"
  >
    <strong
      id="alias"
    >
      Alias:
    </strong>
     
  </div>
  <div
    className="d-flex flex-wrap col m-2  main-details"
  >
    <strong
      id="externalreference"
    >
      External Reference:
    </strong>
     
  </div>
  <ExtraData />
</div>
`;
